tests/test_s3.py

```diff
@@ -3626,3 +3627,40 @@ def run_policy(policy):
         lifecycle = client.get_bucket_lifecycle_configuration(Bucket=bname)
         self.assertEqual(len(lifecycle["Rules"]), 1)
         self.assertEqual(lifecycle["Rules"][0]["ID"], lifecycle_id2)
+
+
+@terraform('aws_s3_encryption_audit')
+def test_s3_encryption_audit(test, aws_s3_encryption_audit):
+    test.patch(s3.S3, "executor_factory", MainThreadExecutor)
+    test.patch(s3.S3, "S3_AUGMENT_TABLE", [])
+
+    session_factory = test.replay_flight_data("test_s3_encryption_audit")
+    p = test.load_policy(
+        {
+            "name": "s3-audit",
+            "resource": "s3",
+            "filters": [
+                {"or": [
+                    {
+                        "type": "bucket-encryption",
+                        "state": False,
+                    },
+                    {
+                        "type": "bucket-encryption",
+                        "crypto": "aws:kms",
+                        "key": "alias/aws/s3",
+                        "state": True,
+                    },
+                    {
+                        "type": "bucket-encryption",
+                        "crypto": "AES256",
+                        "state": True,
+                    },
+                ]},
+            ],
+        },
+        session_factory=session_factory,
+    )
+
+    resources = p.run()
+    test.assertEqual(len(resources), 1)
```

Comment:
we can just do pytest assert len(resources) == 1

we should also verify at least the bucket name matches off the resources list, ie. assert resources['BucketName'] == 'x'